dumper/lib-ext-hive-metastore/build.gradle

```diff
@@ -78,15 +75,25 @@ shadowJar {
 
 }
 
-generateThriftSource {
-    generators {
-        java {
-            out file('build/generated-sources/thrift/main')
-            // option 'java5'
+task generateThrift {
+    def thriftFiles = Files.list(projectDir.toPath().resolve('src/main/thrift'))
+        .filter { it.toString().endsWith('.thrift') }
+        .collect(Collectors.toList())
+    def outputPath = project.projectDir.toPath().resolve("src/main/gen-java")
+    inputs.files(thriftFiles)
+    outputs.dir(outputPath)
+
+    doFirst {
+        project.delete(outputPath)
+        thriftFiles.each { thriftPath ->
+            project.exec {
+                workingDir project.projectDir.toPath().resolve('src/main')
+                commandLine 'thrift', '-r', '--gen', 'java', thriftPath
+            }
         }
     }
 }
 
 sourceSets.main.java {
-    srcDir file('build/generated-sources/thrift/main')
+    srcDir file('src/main/gen-java')
```

Comment:
I think it might be possible (preferable) to say `srcDir generateThrift` here to say "Consume the output of that task, whatever it does."